app/views/handlers/list_collector.py

```diff
@@ -55,5 +69,26 @@ def handle_post(self):
             # wtforms Form parents are not discoverable in the 2.3.3 implementation
             self.questionnaire_store_updater.update_answers(self.form.data)
             self.questionnaire_store_updater.save()
-        else:
+        elif not self.rendered_block.get("repeating_blocks") or self._is_list_collector_complete():
             return super().handle_post()
+
+    def _is_list_collector_complete(self):
+        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
+        return all(
+            self.questionnaire_store_updater.is_section_complete(section_id=self.current_location.section_id, list_item_id=list_item_id)
+            for list_item_id in list_model.items)
+
+    def _get_incomplete_repeating_block_ids(self) -> Generator:
+        repeating_blocks = self.rendered_block.get("repeating_blocks")
+        if not repeating_blocks:
+            return
+
+        list_model = self._questionnaire_store.list_store.get(self.rendered_block["for_list"])
+        for list_item_id in list_model.items:
+            if self.questionnaire_store_updater.is_section_complete(self.current_location.section_id, list_item_id):
+                continue
+
+            complete_block_ids = self.questionnaire_store_updater.get_completed_block_ids(self.current_location.section_id, list_item_id)
+            for repeating_block in repeating_blocks:
+                if repeating_block["id"] not in complete_block_ids:
+                    yield list_item_id, repeating_block["id"]
```

Comment:
Instead of returning a tuple, this should return a `Location` object to be consistent with the rest of the codebase.

Reply:
Secondly, I don't see any place where you are using more than the first incomplete location, therefore to be consistent this should return the first incomplete location, which is consistent with our other router methods called `_get_first_incomplete_location...` etc. This would then just be `get_first_incomplete_repeating_location`.